projects/evm-compatible-bridge/design.md

```diff
@@ -0,0 +1,348 @@
+After the side token is created the bridge contract will associate the contract address with the token uid, making the token supported by the bridge.
+
+With the token supported the admin should make a call to the allowed tokens contract to make it possible to use the token on the bridge.
+
+### Adding support for EVM native tokens
+
+To cross a EVM native token we need to create the equivalent token on Hathor first before allowing it to be crossed.
+
+1. Create a token on Hathor with the same name and symbol.
+1. Melt the tokens created on the transaction above.
+1. Create more authority outputs
+1. Send all authorities to the federation MultiSig address.
+1. Call a method on the bridge contract to map the EVM token to the created token uid.
+1. Call a method on the allowed tokens contract to allow the token to be used on the bridge.
+
+# Alternative solutions
+
+## Granularity
+
+Tokens defined in the ERC-777 have a granularity of 1e18, this means that the smallest amount of tokens that can be transferred is 1e18.
+This is not compatible with the granularity of Hathor tokens, which means we cannot cross 0.0001 tokens to Hathor where the granularity is 1e2.
+
+This limitation means that the bridge can only support sending multiple of 0.01 of any token since it is the higher granularity.
+
+The tokens native to Hathor that cross to the EVM chain can be transactioned in smaller amounts (e.g. 0.0000001 or smaller), but to cross the token back to Hathor we need to send at least 0.01 tokens.
+
+Some other token standards have variable granularity, but if we adhere to the solution that we will only accept multiples of the higher granularity we can support any token standard (when it comes to granularity).
+
+# Future possibilities
+
+Although Hathor's MultiSig has all the secutiry features required to make the bridge functional we still need to rely on external components,
+i.e. a communication channel for actors and a storage for state management (e.g. token equivalency and transaction proposals).
+This could be changed to use a nano-contract, where it can handle events, listing tx proposals and federation with a changing number of participants with 0 downtime.
+Using nano-contracts would also enable very low cost deployment and infrastructure of bridges to new EVM compatible chains and allow for security updates to take effect on all bridges.
+
+# Required features
+
+- We require APIs on the headless wallet to create, mint and melt tokens.
+- We require an API on the headless to return the transaction information with metadata (balance to our wallet, decoded data outputs, etc.).
+- We require a way to start a MultiSig wallet in the lib without a seed or private key.
+  - It should not be a read-only instance since it can send transactions (from gathered signatures)
```

Comment:
It could, I meant that it would not be purely read since it will be the centerpiece to write data to the network.
If by readonly we mean a "seedless" wallet then it will be a readonly wallet.

Reply:
We currently have readonly mode for regular (non-multisig) wallets, right? I guess we should have something similar for multisig, if possible.

Reply:
It is part of the required features and it is under development.